src/deploy/osp_deployer/deployer.py

```diff
@@ -82,16 +97,124 @@ def get_settings():
     return settings, args
 
 
-def run_tempest():
+def run_tempest(director_vm):
     settings, args = get_settings()
     if settings.run_tempest is True or args.run_tempest_only:
         logger.info("=== Running tempest ==")
-        director_vm = Director()
         director_vm.run_tempest()
     else:
         logger.debug("Not running tempest")
 
 
+def deploy_overcloud(director_vm):
+    logger.info("=== Preparing the overcloud ===")
+
+    # The network-environment.yaml must be setup for use during DHCP
+    # server configuration
+    logger.info("Setting up network environment")
+    director_vm.setup_net_envt()
+    logger.info("Setting up dhcp server")
+    director_vm.configure_dhcp_server()
+    logger.info("Discovering nodes")
+    director_vm.node_discovery()
+    logger.info("Configuring iDRACs")
+    director_vm.configure_idracs_core()
+    logger.info("Importing nodes")
+    director_vm.import_nodes()
+    logger.info("Introspecting nodes")
+    director_vm.node_introspection()
+    logger.info("Assigning roles")
+    director_vm.update_sshd_conf()
+    director_vm.assign_node_roles()
+    director_vm.revert_sshd_conf()
+
+    logger.info("Configuring heat templates")
+    director_vm.setup_templates()
+
+    logger.info("=== Installing the overcloud ")
+    logger.debug("installing the overcloud ... this might take a while")
+    director_vm.deploy_overcloud()
+
+
+def deploy_edge(director_vm, node_type):
+    logger.info("=== Installing edge site: %s", str(node_type))
+    director_vm.deploy_edge_site(node_type)
+
+
+def delete_edge(sah_node, director_vm, node_type):
+    logger.info("=== Deleting edge site: %s",
+                node_type)
+    director_vm.delete_edge_site(node_type)
+    logger.info("Cleaning up routes for edge site: {}".format(node_type))
+    director_vm.controller_routes_edge(node_type, False)
+    director_vm.subnet_routes_edge(node_type, False)
+    sah_node.subnet_routes_edge(node_type, False)
+
+
+def get_is_deploy_overcloud():
+    _is_deploy_overcloud = False
+    _input = input("Deploy the overcloud prior to deploying edge site(s)"
+                   ", (y/N)? ")
+    if _input.lower().strip() == "y":
+        _is_deploy_overcloud = True
+    return _is_deploy_overcloud
+
+
+def get_is_deploy_undercloud_and_overcloud():
+    _is_deploy_undercloud = False
+    _input = input("Deploy the undercloud and overcloud prior to "
+                   "deploying edge site(s), (y/N)? ")
+    if _input.lower().strip() == "y":
+        _is_deploy_undercloud = True
+    return _is_deploy_undercloud
```

Comment:
Not a fan of asking for user input .. could we not just add a --edge(s)_only param to be consistant with existing options ?

Reply:
Right now when all you pass in is -s the automation does a full deployment, how should it behave if only -s and --edge_site args are provided,  just assume we are doing a full deployment + an edge site?  Seems dangerous,  the alternative is we make --edge_site mutually exclusive where it fails fast if there is no undercloud and overcloud to add an edge site to.  The detractor being you have to run two commands instead of one, #1 to deploy director/overcloud, #2 to deploy the edge site (--edge_site arg).  I'm all for implementing it this way but wasn't sure how it would go over so I added the prompt.  Please advise.